data race: TestPutRevocationLog

CI run: https://github.com/lightningnetwork/lnd/runs/7008146418?check_suite_focus=true

traceback:

```
WARNING: DATA RACE
Read at 0x000000eb4590 by goroutine 125:
  reflect.Value.Uint()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/value.go:2506 +0xb0d
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:162 +0xa60
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:130 +0x209b
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:127 +0xf79
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:113 +0x1f35
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:130 +0x209b
  reflect.DeepEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:237 +0x3f9
  github.com/stretchr/testify/assert.ObjectsAreEqual()
      /home/runner/work/go/pkg/mod/github.com/stretchr/testify@v1.7.1/assert/assertions.go:65 +0x184
  github.com/stretchr/testify/assert.Equal()
      /home/runner/work/go/pkg/mod/github.com/stretchr/testify@v1.7.1/assert/assertions.go:343 +0x212
  github.com/stretchr/testify/require.Equal()
      /home/runner/work/go/pkg/mod/github.com/stretchr/testify@v1.7.1/require/require.go:162 +0xc8
  github.com/lightningnetwork/lnd/channeldb.TestPutRevocationLog.func2()
      /home/runner/work/lnd/lnd/channeldb/revocation_log_test.go:460 +0x314
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x47
Previous write at 0x000000eb4590 by goroutine 47:
  github.com/lightningnetwork/lnd/channeldb.serializeHTLCEntries()
      /home/runner/work/lnd/lnd/channeldb/revocation_log.go:320 +0xc4
  github.com/lightningnetwork/lnd/channeldb.serializeRevocationLog()
      /home/runner/work/lnd/lnd/channeldb/revocation_log.go:307 +0xba
  github.com/lightningnetwork/lnd/channeldb.TestSerializeRevocationLog()
      /home/runner/work/lnd/lnd/channeldb/revocation_log_test.go:220 +0x224
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x47
Goroutine 125 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x724
  github.com/lightningnetwork/lnd/channeldb.TestPutRevocationLog()
      /home/runner/work/lnd/lnd/channeldb/revocation_log_test.go:451 +0x89e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x47
Goroutine 47 (finished) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x724
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1839 +0x99
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.runTests()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1837 +0x7e4
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1719 +0xa71
  github.com/lightningnetwork/lnd/kvdb.RunTests()
      /home/runner/work/go/pkg/mod/github.com/lightningnetwork/lnd/kvdb@v1.3.1/test_utils.go:23 +0x2e
  github.com/lightningnetwork/lnd/channeldb.TestMain()
      /home/runner/work/lnd/lnd/channeldb/setup_test.go:10 +0x318
  main.main()
      _testmain.go:333 +0x313
==================
==================
--- FAIL: TestSentPaymentSerialization (0.00s)
    testing.go:1312: race detected during execution of test
WARNING: DATA RACE
Read at 0x000000eb4598 by goroutine 125:
  reflect.Value.Uint()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/value.go:2500 +0xa54
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:162 +0xa60
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:130 +0x209b
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:127 +0xf79
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:113 +0x1f35
  reflect.deepValueEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:130 +0x209b
  reflect.DeepEqual()
      /opt/hostedtoolcache/go/1.18.2/x64/src/reflect/deepequal.go:237 +0x3f9
  github.com/stretchr/testify/assert.ObjectsAreEqual()
      /home/runner/work/go/pkg/mod/github.com/stretchr/testify@v1.7.1/assert/assertions.go:65 +0x184
  github.com/stretchr/testify/assert.Equal()
      /home/runner/work/go/pkg/mod/github.com/stretchr/testify@v1.7.1/assert/assertions.go:343 +0x212
  github.com/stretchr/testify/require.Equal()
      /home/runner/work/go/pkg/mod/github.com/stretchr/testify@v1.7.1/require/require.go:162 +0xc8
  github.com/lightningnetwork/lnd/channeldb.TestPutRevocationLog.func2()
      /home/runner/work/lnd/lnd/channeldb/revocation_log_test.go:460 +0x314
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x47
Previous write at 0x000000eb4598 by goroutine 47:
  github.com/lightningnetwork/lnd/channeldb.serializeHTLCEntries()
      /home/runner/work/lnd/lnd/channeldb/revocation_log.go:316 +0x98
  github.com/lightningnetwork/lnd/channeldb.serializeRevocationLog()
      /home/runner/work/lnd/lnd/channeldb/revocation_log.go:307 +0xba
  github.com/lightningnetwork/lnd/channeldb.TestSerializeRevocationLog()
      /home/runner/work/lnd/lnd/channeldb/revocation_log_test.go:220 +0x224
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x47
Goroutine 125 (running) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x724
  github.com/lightningnetwork/lnd/channeldb.TestPutRevocationLog()
      /home/runner/work/lnd/lnd/channeldb/revocation_log_test.go:451 +0x89e
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.(*T).Run.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x47
Goroutine 47 (finished) created at:
  testing.(*T).Run()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1486 +0x724
  testing.runTests.func1()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1839 +0x99
  testing.tRunner()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1439 +0x213
  testing.runTests()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1837 +0x7e4
  testing.(*M).Run()
      /opt/hostedtoolcache/go/1.18.2/x64/src/testing/testing.go:1719 +0xa71
  github.com/lightningnetwork/lnd/kvdb.RunTests()
      /home/runner/work/go/pkg/mod/github.com/lightningnetwork/lnd/kvdb@v1.3.1/test_utils.go:23 +0x2e
  github.com/lightningnetwork/lnd/channeldb.TestMain()
      /home/runner/work/lnd/lnd/channeldb/setup_test.go:10 +0x318
  main.main()
      _testmain.go:333 +0x313
==================
--- FAIL: TestPutRevocationLog (0.05s)
    --- FAIL: TestPutRevocationLog/dust_htlc_not_saved (0.00s)
        testing.go:1312: race detected during execution of test
    testing.go:1312: race detected during execution of test
FAIL
FAIL	github.com/lightningnetwork/lnd/channeldb	11.414s
FAIL
```

--- a/channeldb/revocation_log_test.go
+++ b/channeldb/revocation_log_test.go
@@ -211,9 +211,10 @@ func TestSerializeHTLCEntries(t *testing.T) {
 func TestSerializeRevocationLog(t *testing.T) {
 	t.Parallel()
 
-	// Copy the testRevocationLog.
+	// Copy the testRevocationLog and testHTLCEntry.
 	rl := testRevocationLog
-	rl.HTLCEntries = []*HTLCEntry{&testHTLCEntry}
+	htlc := testHTLCEntry
+	rl.HTLCEntries = []*HTLCEntry{&htlc}
 
 	// Write the tlv stream.
 	buf := bytes.NewBuffer([]byte{})

--- a/docs/release-notes/release-notes-0.15.1.md
+++ b/docs/release-notes/release-notes-0.15.1.md
@@ -15,8 +15,14 @@
 * [Add previous_outpoints to listchaintxns](https://github.com/lightningnetwork/lnd/pull/6321)
 
 
+## Bug Fixes
+
+* Fixed data race found in
+  [`TestSerializeHTLCEntries`](https://github.com/lightningnetwork/lnd/pull/6673).
+
 # Contributors (Alphabetical Order)
 
 * Elle Mouton
 * ErikEk
 * Priyansh Rastogi
+* Yong Yu

